honeybadgermpc/hbavss_batch.py

```diff
@@ -0,0 +1,235 @@
+        self.n, self.t, self.my_id = n, t, my_id
+        self.g = g
+        self.poly_commit = PolyCommit(g, h)
+
+        # Create a mechanism to split the `recv` channels based on `tag`
+        self.subscribe_recv_task, self.subscribe_recv = subscribe_recv(recv)
+
+        # Create a mechanism to split the `send` channels based on `tag`
+        def _send(tag):
+            return wrap_send(tag, send)
+        self.get_send = _send
+
+        # This is added to consume the share the moment it is generated.
+        # This is especially helpful when running multiple AVSSes in parallel.
+        self.output_queue = asyncio.Queue()
+
+        self.field = ZR
+        self.poly = polynomials_over(self.field)
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, type, value, traceback):
+        self.subscribe_recv_task.cancel()
+
+    async def _process_avss_msg(self, avss_id, dealer_id, dispersal_msg):
+        tag = f"{dealer_id}-{avss_id}-AVSS"
+        send, recv = self.get_send(tag), self.subscribe_recv(tag)
+
+        def multicast(msg):
+            for i in range(self.n):
+                send(i, msg)
+
+        commitments, ephemeral_public_key, encrypted_witnesses = loads(
+            dispersal_msg)
+
+        # # all_encrypted_witnesses: n
+        shared_key = pow(ephemeral_public_key, self.private_key)
+
+        shares = [None] * (self.t+1)
+        witnesses = [None] * (self.t+1)
+        # Decrypt
+        for k in range(self.t+1):
+            # print(self.my_id, k, len(encrypted_witnesses[self.my_id * (self.t+1) + k]))
+            shares[k], witnesses[k] = SymmetricCrypto.decrypt(
+                str(shared_key).encode("utf-8"), encrypted_witnesses[self.my_id * (self.t+1) + k])
+
+        ##################### Don't know why this doesn't work #####################
+        # # Decrypt
+        # for k in range(self.t+1):
+        #     shares[k], witnesses[k] = SymmetricCrypto.decrypt(
+        #         str(shared_key).encode("utf-8"), encrypted_witnesses[self.my_id][k])
+        ############################################################################
+
+        # verify & send all ok
+        for i in range(self.t+1):
+            if self.poly_commit.verify_eval(commitments[i], self.my_id+1, shares[i], witnesses[i]):
+                multicast(HbAVSSMessageType.OK)
+            else:
+                multicast(HbAVSSMessageType.IMPLICATE)
+
+        oks_recvd = 0
+        readys_recvd = 0
+        # if 2t+1 OK or t+1 READY -> send all ready
+        while oks_recvd < 2 * self.t + 1 and readys_recvd < self.t + 1:
+            _, avss_msg = await recv()  # First value is the `sid`
+            if avss_msg == HbAVSSMessageType.OK:
+                oks_recvd += 1
+            if avss_msg == HbAVSSMessageType.READY:
+                readys_recvd += 1
+
+        # logging.debug("[%d][HbAvssBatch] 2t+1 OKs or t+1 READYs received.", self.my_id)
```

Comment:
Is there any reason why you want to comment the logs? Please either uncomment them or delete them?

Reply:
Resolved.